mvn-resolver-transport-http3/src/test/java/com/artipie/aether/transport/http3/ArtipieHTTP3IT.java

```diff
@@ -0,0 +1,103 @@
+ * 1) {@link ArtipieHTTP3IT#artipie} is container with latest artipie version running on ubuntu 22.04 and java 21<p>
+ * 2) {@link ArtipieHTTP3IT#mavenClient} is ubuntu 22.04 based container with this plugin built with maven 3.9.5 under java 21,
+ * see ./Dockerfile<p>
+ *
+ * The containers are connected via docker {@link Network} with alias 'artipie'. Artipie container
+ * runs HTTP3 maven-proxy repository available at <code>https://artipie:8091/my-maven-proxy/</code>.
+ * <p>
+ * Test maven project {@code ./resources/com/example/maven-http3} which uses this plugin is added to
+ * {@link ArtipieHTTP3IT#mavenClient} and built with maven settings
+ * {@code ./resources/com/example/maven-http3/maven-settings.xml}. The dependencies of this project
+ * are obtained from Artipie via HTTP3.
+ *
+ */
+public class ArtipieHTTP3IT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ArtipieHTTP3IT.class);
+
+    private GenericContainer<?> mavenClient;
+
+    private GenericContainer<?> artipie;
+
+    private final Consumer<OutputFrame> artipieLog =
+        new Slf4jLogConsumer(LoggerFactory.getLogger(this.getClass())).withPrefix("ARTIPIE");
+
+    private Network net;
+
+    @BeforeEach
+    void init() throws IOException, InterruptedException {
+        this.net = Network.newNetwork();
+        this.artipie =  new GenericContainer<>("artipie/artipie-ubuntu:latest")
+            .withNetwork(this.net)
+            .withNetworkAliases("artipie")
+            .withLogConsumer(this.artipieLog)
+            .withClasspathResourceMapping(
+                "artipie.yaml", "/etc/artipie/artipie.yml", BindMode.READ_ONLY
+            )
+            .withClasspathResourceMapping(
+                "my-maven-proxy.yaml", "/var/artipie/repo/my-maven-proxy.yaml", BindMode.READ_ONLY
+            )
+            .withWorkingDirectory("/w");
+        this.mavenClient = new GenericContainer<>("http3-resolver:1.0-SNAPSHOT")
+            .withNetwork(this.net)
+            .withCommand("tail", "-f", "/dev/null")
+            .withWorkingDirectory("/w");
+        this.mavenClient.start();
+        this.artipie.start();
+        this.mavenClient.execInContainer("sleep", "5");
+    }
+
+    @Test
+    void resolvesDependencies() throws IOException, InterruptedException {
+        this.putClasspathResourceToClient("com/example/maven-http3/maven-settings.xml", "/w/settings.xml");
+        this.putClasspathResourceToClient("com/example/maven-http3/pom.xml", "/w/pom.xml");
+        final Container.ExecResult exec = this.mavenClient.execInContainer(
+            "mvn", "install", "-s", "settings.xml", "-Daether.connector.https.securityMode=insecure"
+        );
+        String res = String.join("\n", exec.getStdout(), exec.getStderr());
+        LOGGER.info(res);
+        MatcherAssert.assertThat("Maven install is not successful", exec.getExitCode() == 0);
+        MatcherAssert.assertThat(
+            res,
+            Matchers.stringContainsInOrder(
```

Comment:
How can it prove that the packages were gotten by HTTP 3?

Reply:
a) `https://artipie:8091/my-maven-proxy` - is a path to artipie http3 repository
b) is such configuration maven client can perform request over http3 only (it fails trying to get dependency from central over http 1)

So there are only 2 possibilities in current configuration: (either build is success and deps were obtained via http3) or it fails